django/db/models/base.py

```diff
@@ -720,12 +720,13 @@ def refresh_from_db(self, using=None, fields=None, from_queryset=None):
         if fields is not None:
             db_instance_qs = db_instance_qs.only(*fields)
         elif deferred_fields:
-            fields = {
```

Comment:
This was the main issue: this branch reassigns `fields`. Until the previous fix in 73df8b54a2fab53bec4c7573cda5ad8c869c2fd8, this reassigned `fields` wasn't used. Then it started using it, even though that didn't make sense - `attname` is not the same as a field's name.